pkg/metrics/storage/storage_sampler.go

```diff
@@ -393,16 +394,23 @@ func populateUsage(fsUsage *disk.UsageStat, dest *Sample) {
 	totalBytes := PlatformFsByteScale(fsUsage.Total)
 	freeBytes := PlatformFsByteScale(fsUsage.Free)
 
-	dest.UsedBytes = &usedBytes
-	dest.TotalBytes = &totalBytes
-	dest.FreeBytes = &freeBytes
+	dest.UsedBytes = asValidFloatPtr(&usedBytes)
+	dest.TotalBytes = asValidFloatPtr(&totalBytes)
+	dest.FreeBytes = asValidFloatPtr(&freeBytes)
 
 	// used percent calculations use total of usedBytes + freeBytes since totalBytes
 	// on linux includes space reserved for the operating system
 	usedPercent := usedBytes / (usedBytes + freeBytes) * 100
 	freePercent := 100 - usedPercent
-	dest.UsedPercent = &usedPercent
-	dest.FreePercent = &freePercent
+	dest.UsedPercent = asValidFloatPtr(&usedPercent)
+	dest.FreePercent = asValidFloatPtr(&freePercent)
 
 	populateUsageOS(fsUsage, dest)
 }
+
+func asValidFloatPtr(value *float64) *float64 {
+	if value == nil || math.IsNaN(*value) || math.IsInf(*value, 0) {
```

Comment:
Reporting a nil for the metrics normally means no value was collected. We should check for consistency what other newrelic collectors does for +Inf/-Inf. For e.g. nrjmx reports the maxFloatValue